qemu/tests/live_backup_multi_block.py

```diff
@@ -0,0 +1,189 @@
+
+    def generate_params(self):
+        """
+        Generate params set for all images, including the image_chain and
+        the transaction command lists for dirty bitmap and backup jobs.
+        """
+        for image in self.image_list:
+            full_backup_image = "%sFullbackup" % image
+            increm_backup_image = "%sIncremental0" % image
+            self.params["image_name_%s" % increm_backup_image] = (
+                "images/%s" % increm_backup_image)
+            self.image_chain_map[image] = (
+                "%s %s" % (full_backup_image, increm_backup_image))
+
+            # generate params for full backup images.
+            image_params = self.image_chain_reassign(image)
+            self.generate_backup_params()
+
+            # generate bitmap transaction args.
+            image_file = qemu_storage.storage.get_image_filename(
+                image_params, self.data_dir)
+            drive_name = self.vm.get_block({"file": image_file})
+            bitmap_name = "bitmap_%s" % image
+            self.bitmap_list.append(bitmap_name)
+            bitmap_args = {"node": drive_name, "name": bitmap_name}
+            self.transaction_add(self.bitmap_args_list,
+                                 "block-dirty-bitmap-add", bitmap_args)
+
+            # generate full backup transaction args.
+            backup_format = self.backup_format
+            backup_image_name = "images/%s.%s" % (full_backup_image,
+                                                  backup_format)
+            backup_image_name = utils_misc.get_path(
+                self.data_dir, backup_image_name)
+            full_backup_args = {"device": drive_name,
+                                "target": backup_image_name,
+                                "format": backup_format,
+                                "sync": "full"}
+            self.transaction_add(self.full_backup_arg_list,
+                                 "drive-backup", full_backup_args)
+
+            # generate incremental backup transaction args.
+            backup_image_name = "images/%s.%s" % (increm_backup_image,
+                                                  backup_format)
+            backup_image_name = utils_misc.get_path(
+                self.data_dir, backup_image_name)
+            backup_format = self.backup_format
+            increm_backup_args = {"device": drive_name,
+                                  "target": backup_image_name,
+                                  "format": backup_format,
+                                  "sync": "incremental",
+                                  "mode": "existing",
+                                  "bitmap": bitmap_name}
+            self.transaction_add(self.increm_backup_arg_list,
+                                 "drive-backup", increm_backup_args)
+
+    def create_backup_images(self):
+        """
+        Create backup for all images.
+        """
+        for image in self.image_list:
+            self.backup_index = 1
```

Comment:
@sitoliu Yes, it is intended. It will be added up each time calling self.create_backup_image(), so must reset to 1 for each image in this loop.